### ISSUE TITLE
Login and personal_urls

### ISSUE THREAD

**Issue by paulb-smartit** [Hutchy68/pivot#99]
### Setup

- Browsers and versions: Firefox 73.0.1
- MediaWiki version: 1.34.2
- Database version: 10.3.23-MariaDB
- PHP version: 7.3.19 (apache2handler)
- Pivot version: 2.2.0
- Extension version (if connected to the issue):  MW-OAuth2Client | 0.4 (adf8e77) 03:26, 23 July 2020

### Issue

Detailed description of the issue goes here.

The extension MW-OAuth2Client places a link to the OAuth Login page into the personal_urls array. This link doesn't appear in the pivot page anywhere to be used to login. The only personal_urls item displayed is Login.

```php
$personal_urls['anon_oauth_login']['href']
```

This maybe an issue with the extension, but any guidance you can give on this would be appreciated.

**Comment by Hutchy68** [comment]
Yes, I suppose there was a flaw in my logic of creating only a login method that was MW centric. So try this...

Change... `Pivot.skin.php` @ lines 132-147 from

```
					<aside class="right-off-canvas-menu">
					  <ul class="off-canvas-list">
					<?php if ($wgUser->isLoggedIn()): ?>
						<li id="personal-tools"><label>Personal</label></li>
						<?php foreach ($this->getPersonalTools() as $key => $item) { echo $this->makeListItem($key, $item); } ?>
							<?php else: ?>
								<?php if (isset($this->data['personal_urls']['anonlogin'])): ?>
									<li><a href="<?php echo $this->data['personal_urls']['anonlogin']['href']; ?>"><?php echo wfMessage( 'login' )->text() ?></a></li>
								<?php elseif (isset($this->data['personal_urls']['login'])): ?>
									<li><a href="<?php echo htmlspecialchars($this->data['personal_urls']['login']['href']); ?>"><?php echo wfMessage( 'login' )->text() ?></a></li>
										<?php else: ?>
											<li><?php echo Linker::link(Title::newFromText('Special:UserLogin'), wfMessage( 'login' )->text()); ?></li>
								<?php endif; ?>
							<?php endif; ?>
					  </ul>
					</aside>
```

```
					<aside class="right-off-canvas-menu">
					  <ul class="off-canvas-list">
						<li id="personal-tools"><label>Personal</label></li>
						<?php foreach ($this->getPersonalTools() as $key => $item) { echo $this->makeListItem($key, $item); } ?>
					  </ul>
					</aside>
```

If you can wait, I will push a patch. I think there can be some more improvements and the title Personal under `personal-tools` should be a message. It probably should change from Login/Create Account to Personal Tools when logged in. Any other ideas or suggestions on this appreciated from anyone.

**Comment by Hutchy68** [comment]
Use git to fetch and then checkout branch `login-improvements`

Comparison of https://github.com/Hutchy68/pivot/compare/develop...login-improvements which includes a fix for Issue #98 

Please comment on the fixes so I can prep an update and new release. Thanks!

**Comment by Hutchy68** [comment]
Merged into `develop` branch. No replies on issue, so closing. Thanks for reporting.
